source/Tests/Parsing/ParserFixture.cs

```diff
@@ -0,0 +1,34 @@
+using FluentAssertions;
+using NUnit.Framework;
+using Octopus.Ocl;
+using Octopus.Ocl.Parsing;
+
+namespace Tests.Parsing
+{
+    public class ParserFixture
+    {
+        [Test]
+        public void Empty()
+            => OclParser.Execute("")
+                .Should()
+                .Be(new OclDocument());
+
+        [Test]
+        public void Blank()
+            => OclParser.Execute(" \t \r\n \t ")
+                .Should()
+                .Be(new OclDocument());
+
+        [Test]
+        public void InvalidDocumentReportsCorrectLineNumber()
+        {
+            var (_, error) = OclParser.TryExecute(@"
+                    Foo = ""Bar""
+
+                    Foo = ""Bar
+            ");
+
+            error.Should().Be("Parsing failure: unexpected 'F'; expected end of input (Line 4, Column 21); recently consumed:           ");
```

Comment:
We _might_ be able to make this error location better if we use `X` versions of combinators in the right places. Maybe. If that doesn't fix it, we would have to look at something more advanced like https://www.nuget.org/packages/Superpower/